src/dmd/expressionsem.d

```diff
@@ -11399,6 +11399,12 @@ Expression semanticX(DotIdExp exp, Scope* sc)
                     {
                         return new ErrorExp();
                     }
+                    if (f.flags & (FUNCFLAG.purityInprocess | FUNCFLAG.safetyInprocess |
```

Comment:
I think that this will not suffice for nested functions:

```d
void func(T)
{
    void foo()
    {
        pragma(msg, foo.mangleof);
    }
}
```